server/game/core/card/Card.ts

```diff
@@ -434,17 +443,30 @@ export class Card extends OngoingEffectSource {
 
 
     // ******************************************* ZONE MANAGEMENT *******************************************
-    public moveTo(targetZone: ZoneName) {
+    public moveTo(targetZone: MoveZoneName) {
+        Contract.assertNotNullLike(this._zone, `Attempting to move card ${this.internalName} before initializing zone`);
+
         const originalZone = this.zoneName;
 
         if (originalZone === targetZone) {
             return;
         }
 
         this.cleanupBeforeMove(targetZone);
-        const prevZone = this._zoneName;
-        this._zoneName = targetZone;
-        this.initializeForCurrentZone(prevZone);
+
+        const prevZone = this._zone;
```

Comment:
Are there any effects/cards like Gar Saxon that would want us to store the previous zone on the card after it has moved?

Reply:
Yeah, for defeat purposes we'll need that. But that's kind of a separate task, haven't really figured out how that will be stored yet